SCons/Environment.py

```diff
@@ -1250,9 +1250,11 @@ def __init__(
         SCons.Tool.Initializers(self)
 
         if tools is None:
-            tools = self._dict.get('TOOLS', None)
-            if tools is None:
-                tools = ['default']
+            tools = self._dict.get('TOOLS', ['default'])
+        else:
+            # for a new env, if we didn't use TOOLS, make sure it starts empty
+            # so it only shows tools actually initialized.
+            self._dict['TOOLS'] = []
```

Comment:
This is new behavoir.  There's no reason you can't set tools via `tools` and `TOOLS` when calling Environment().. ?

Reply:
Well, you can't currently. If you want to make that work, that's new behavior too.
```py
        if tools is None:                                                      
            tools = self._dict.get('TOOLS', None)                              
            if tools is None:                                                  
                tools = ['default'] 
```
That is, don't look at `TOOLS` unless `tools` was not given and thus is the default `None`.

Reply:
> Well, you can't currently. 

sloppy wording: currently you can set both, but if you set `tools` it does't matter if you set `TOOLS`.

Reply:
Are you sure? From the code I was looking at it ends up calling `tool(env)` so env['TOOLS'] is never cleared, just appended to.

Reply:
Yes, that's actually the point.  You call the tool for each tool specified. If `Environment` got called with anything, those are the tools called. In that case, `Environment` doesn't use `TOOLS`, and so any contents of the variable are retained, but the retained tools don't reflect tools that were called, so the information is either deceptive, or duplicate, depending on what was in `tools`. It doesn't hurt anything except to confuse humans, since the information in `TOOLS` basically isn't used afterward.

Reply:
see: https://github.com/SCons/scons/blob/master/SCons/EnvironmentTests.py#L1424

There's a test for this working.. at least setting them in the DefaultEnvironment() and expecting them to get initialized in subsequent Environment()'s when tools is not specified...

Reply:
Of course, another approach is to read `TOOLS` if set and add it to `tools` if it was set.  That's not what I was proposing, just to make the current reporting (aka eventual value of `TOOLS`) accurate wrt  how it is implemented now.

Reply:
> see: https://github.com/SCons/scons/blob/master/SCons/EnvironmentTests.py#L1424
> 
> There's a test for this working.. at least setting them in the DefaultEnvironment() and expecting them to get initialized in subsequent Environment()'s when tools is not specified...

That's not going to be broken - if `tools` is not specified, then `TOOLS` is used, and is not cleared.

Reply:
Ahh. ok. you're correct.. I think I need some coffee.. Thanks for your patience.

Reply:
I think we've passed the Coffee Line...